app/models/course/material/folder.rb

```diff
@@ -3,6 +3,8 @@ class Course::Material::Folder < ActiveRecord::Base
   include Course::ModelComponentHost::Component
 
   after_initialize :set_defaults, if: :new_record?
+  before_validation :assign_valid_name
+  before_save :assign_valid_name
```

Comment:
Any circumstance where you need to run the callback in both before_validation and before_safe?